plugins/woocommerce/src/Admin/Features/MultichannelMarketing/Init.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Multichannel Marketing Experience
+ *
+ * @package Woocommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\MultichannelMarketing;
+
+use Automattic\WooCommerce\Internal\Admin\Survey;
+use Automattic\WooCommerce\Admin\Features\Features;
+use Automattic\WooCommerce\Internal\Admin\WCAdminAssets;
+
+/**
+ * Contains logic for Multichannel Marketing.
+ */
+class Init {
+	/**
+	 * Option name used to toggle this feature.
+	 */
+	const TOGGLE_OPTION_NAME = 'woocommerce_multichannel_marketing_enabled';
+
+	/**
+	 * Determines if the feature has been toggled on or off.
+	 *
+	 * @var boolean
+	 */
+	protected static $is_updated = false;
+
+	/**
+	 * Hook into WooCommerce.
+	 */
+	public function __construct() {
+		add_filter( 'woocommerce_settings_features', array( $this, 'add_feature_toggle' ) );
+		add_action( 'update_option_' . self::TOGGLE_OPTION_NAME, array( $this, 'reload_page_on_toggle' ), 10, 2 );
+		add_action( 'woocommerce_settings_saved', array( $this, 'maybe_reload_page' ) );
+	}
+
+	/**
+	 * Add the feature toggle to the features settings.
+	 *
+	 * @param array $features Feature sections.
+	 * @return array
+	 */
+	public static function add_feature_toggle( $features ) {
+		$description = __(
+			'Enables the new WooCommerce Multichannel Marketing experience in the Marketing page',
+			'woocommerce'
+		);
+
+		$features[] = array(
+			'title' => __( 'Marketing', 'woocommerce' ),
+			'desc'  => $description,
+			'id'    => self::TOGGLE_OPTION_NAME,
+			'type'  => 'checkbox',
+			'class' => '',
+		);
+
+		return $features;
+	}
+
+	/**
+	 * Reloads the page when the option is toggled to make sure all Multichannel Marketing features are loaded.
+	 *
+	 * @param string $old_value Old value.
+	 * @param string $value     New value.
+	 */
+	public static function reload_page_on_toggle( $old_value, $value ) {
+		if ( $old_value === $value ) {
+			return;
+		}
+
+		self::$is_updated = true;
```

Comment:
```suggestion
		if ( $old_value !== $value ) {
			self::$is_updated = true;
		}
```

Reply:
The function here is inspired from the existing code below:

https://github.com/woocommerce/woocommerce/blob/26df595313761ce264cb7db7a5ad9e2d5a834892/plugins/woocommerce/src/Admin/Features/Navigation/Init.php#L117-L127

https://github.com/woocommerce/woocommerce/blob/26df595313761ce264cb7db7a5ad9e2d5a834892/plugins/woocommerce/src/Internal/Admin/Analytics.php#L97-L103

I'm thinking it may be better to stick with the "return early" pattern in the existing code base.